### PR TITLE
Fix "Error handling issue"

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-77846
 - CVS-77850

### CODE REVIEW

File: src/core/src/node.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ ov::Node::Node(const OutputVector& arguments, size_t output_size) : Node() {
     set_output_size(output_size);
 }
 
-ov::Node::~Node() {
+ov::Node::~Node() noexcept(false) {
```
Comment (ilyachur) [inline src/core/src/node.cpp:77]: Destructor cannot throw exceptions

PR with the same changes: https://github.com/openvinotoolkit/openvino/pull/10123

File: src/common/legacy/src/cnn_network_impl.cpp [hunk 2]
```diff
@@ -349,7 +349,7 @@ void CNNNetworkImpl::removeOutput(const string& dataName) {
     }
 }
 
-size_t CNNNetworkImpl::getBatchSize() const noexcept {
+size_t CNNNetworkImpl::getBatchSize() const {
```
Comment (ilyachur) [inline src/common/legacy/src/cnn_network_impl.cpp:352]: Why did you remove noexcept for all methods?
at() throws an exception? Or something else?

**Comment by vurusovs** [comment]
Closing PR in favor of https://github.com/openvinotoolkit/openvino/pull/10123

File: src/common/legacy/src/cnn_network_impl.cpp (hunk 2 above)
Comment (vurusovs) [reply src/common/legacy/src/cnn_network_impl.cpp]: There was a false positive from algorithm perspective - coverity shows the case which causes exception, but it was handled above in a code. Reverted

File: src/core/src/node.cpp (hunk 1 above)
Comment (vurusovs) [reply src/core/src/node.cpp]: Reverted

File: src/inference/src/cnn_network_ngraph_impl.cpp [hunk 3]
```diff
@@ -358,45 +358,55 @@ void CNNNetworkNGraphImpl::reshape() {
...
+        return OK;
+    } catch (const InferenceEngine::GeneralError&) {
+        return GENERAL_ERROR;
+    } catch (const ov::Exception&) {
+        return GENERAL_ERROR;
+    } catch (const std::out_of_range&) {
+        return OUT_OF_BOUNDS;
+    } catch (const std::runtime_error&) {
+        return GENERAL_ERROR;
+    }
```
Comment (ilyachur) [inline src/inference/src/cnn_network_ngraph_impl.cpp:409]: Do we have all these exception types? Can we also add the exception message?

Comment (vurusovs) [reply src/inference/src/cnn_network_ngraph_impl.cpp]: All the exceptions may be thrown from `reshape()`. Added exception message handling

**Review by ilyachur — APPROVED** [review]
